examples/ogbn_papers_100m.py

```diff
@@ -23,51 +54,91 @@ def get_num_workers() -> int:
-optimizer = torch.optim.Adam(model.parameters(), lr=0.01, weight_decay=0.0005)
+# Set Up Neighbor Loading
+data = dataset[0]
+if args.cugraph_data_loader:
+    import cupy
+    import rmm
+
+    rmm.reinitialize(devices=[0], pool_allocator=True, initial_pool_size=78e9,
+                     managed_memory=True)
+
+    from rmm.allocators.torch import rmm_torch_allocator
+    torch.cuda.memory.change_current_allocator(rmm_torch_allocator)
+
+    from rmm.allocators.cupy import rmm_cupy_allocator
+    cupy.cuda.set_allocator(rmm_cupy_allocator)
+
+    import cugraph
+    from cugraph_pyg.data import CuGraphStore
+    from cugraph_pyg.loader import CuGraphNeighborLoader
+    G = {("N", "E", "N"): data.edge_index}
+    N = {"N": data.num_nodes}
+    fs = cugraph.gnn.FeatureStore(backend="torch")
+    fs.add_data(data.x, "N", "x")
+    fs.add_data(data.y, "N", "y")
+    cugraph_store = CuGraphStore(fs, G, N)
+    train_loader = CuGraphNeighborLoader(cugraph_store,
+                                         input_nodes=split_idx['train'],
+                                         shuffle=True, drop_last=True,
+                                         **kwargs)
+    val_loader = CuGraphNeighborLoader(cugraph_store,
+                                       input_nodes=split_idx['valid'],
+                                       **kwargs)
+    test_loader = CuGraphNeighborLoader(cugraph_store,
+                                        input_nodes=split_idx['test'],
+                                        **kwargs)
+else:
+    num_work = get_num_workers()
+    train_loader = NeighborLoader(data=data, input_nodes=split_idx['train'],
+                                  num_workers=num_work, drop_last=True,
+                                  shuffle=False, **kwargs)
+    val_loader = NeighborLoader(data=data, input_nodes=split_idx['valid'],
+                                num_workers=num_work, **kwargs)
+    test_loader = NeighborLoader(data=data, input_nodes=split_idx['test'],
+                                 num_workers=num_work, **kwargs)
+
+if args.use_gat_conv:
+    model = torch_geometric.nn.models.GAT(
+        dataset.num_features, args.hidden_channels, args.num_layers,
+        dataset.num_classes, heads=args.n_gat_conv_heads).to(device)
+else:
+    model = torch_geometric.nn.models.GCN(dataset.num_features,
+                                          args.hidden_channels,
+                                          args.num_layers,
+                                          dataset.num_classes).to(device)
+
+optimizer = torch.optim.Adam(model.parameters(), lr=args.lr,
+                             weight_decay=0.0005)
+
+warmup_steps = 20
 
 
 def train():
     model.train()
-
     for i, batch in enumerate(train_loader):
-        start = time.perf_counter()
+        if isinstance(batch, torch_geometric.data.HeteroData):
+            batch = batch.to_homogeneous()
+
+        if i == warmup_steps:
+            start_avg_time = time.perf_counter()
```

Comment:
Looks like this is not average time but per-iter time. You might need to rerun your experiments if your results are based on this time counter.

Reply:
im a bit confused. the timer starts once we have finished warming up, then at the end of the epoch we check the time elapsed and divide by the number of iters passed since the timer starts. this gives us an average time for each iteration which is what we report